### PR TITLE
Contentful ui integration

### PR DESCRIPTION
**Pull request by piekczyk**
# [Contentful ui integration](https://app.shortcut.com/oazo-apps/story/13987/landing-page-data-and-ui-integration?vc_group_by=day&ct_workflow=all&cf_workflow=500000053)

plus also: https://app.shortcut.com/oazo-apps/story/13986/create-api-endpoint-for-single-landing-page
<please insert a shortcut link above>
  
## Changes 👷‍♀️
  <Please add short list of changes>

- integrated contentful api
  
## How to test 🧪
  <Please explain how to test your changes>

- `better-on-summer/[slug]`

### CODE REVIEW

File: pages/better-on-summer/[...slug].tsx [hunk 1]
```diff
@@ -25,213 +22,207 @@ function MarketingTemplatePage(props: MarketingTemplatePageProps) {
...
-              ['Active Earn Strategy available', true, true, true, false],
-              ['Multi Collateral Positions', true, false, true, false],
-            ],
-          },
-        ],
+export async function getServerSideProps({ locale, params }: GetServerSidePropsContext) {
+  const slug = params?.slug
+  if (!slug) {
+    return {
+      redirect: {
```
Comment (piotrkonowrocki) [inline pages/better-on-summer/[...slug].tsx:29]: Shouldn't there be
```
   return {
     notFound: true,
   }
```
?

File: pages/better-on-summer/[...slug].tsx [hunk 2]
```diff
@@ -25,213 +22,207 @@ function MarketingTemplatePage(props: MarketingTemplatePageProps) {
...
+  const resolvedSlug = Array.isArray(slug) ? slug.join('/') : slug
+
+  const { palette, hero, blocks } = await getLandingPageBySlug(resolvedSlug)
+
+  try {
+    const marketingTemplatePageProps: MarketingTemplateFreeform = {
+      palette,
+      hero,
+      blocks,
+      // blocks: [
```
Comment (piotrkonowrocki) [inline pages/better-on-summer/[...slug].tsx:45]: Could be just removed, we don't need it anymore.

**Review by piotrkonowrocki — CHANGES_REQUESTED** [review]

File: pages/better-on-summer/[...slug].tsx (hunk 2 above)
Comment (piekczyk) [reply pages/better-on-summer/[...slug].tsx:45]: fixed

File: contentful/queries/getLandingPageBySlug.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+      title: lendingPost.hero.title,
+      description: lendingPost.hero.description,
+      link: lendingPost.hero.link,
+      image: lendingPost.hero.image.url,
+    },
+    blocks: mapBlocksCollection(lendingPost.blocksCollection, entryCollection),
+  }
+}
+
+export async function getLandingPageBySlug(slug: string | undefined) {
```
[the remark this answers is not in the document]
Comment (piekczyk) [reply contentful/queries/getLandingPageBySlug.ts:37]: fixed

File: contentful/mappers/mapLandingInfoBlock.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+export const mapLandingInfoBlock = (blockItem: LandingPageInfo): MarketingTemplateInfoBoxBlock => ({
+  type: MarketingTemplateBlocks.INFO_BOX,
+  title: blockItem.title,
+  // description: blockItem.description, // TODO
+  description: 'zelipapo',
+  content: blockItem.collection.map((item) => ({
+    title: item.title,
+    // description: item.description,
+    description: 'zelipapo', // TODO
```
[the remark this answers is not in the document]
Comment (piekczyk) [reply contentful/mappers/mapLandingInfoBlock.ts]: fixed

File: contentful/mappers/mapLandingPageBannerBlock.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import type { MarketingTemplateBannerBlock } from 'features/marketing-layouts/types'
+import { MarketingTemplateBlocks } from 'features/marketing-layouts/types'
+
+export const mapLandingPageBannerBlock = (
+  blockItem: LandingPageBanner,
+): MarketingTemplateBannerBlock => ({
+  type: MarketingTemplateBlocks.BANNER,
+  content: blockItem.collection.map((item) => ({
+    title: item.title,
+    // description: item.description?.json, // TODO
```
[the remark this answers is not in the document]
Comment (piekczyk) [reply contentful/mappers/mapLandingPageBannerBlock.ts]: fixed

File: contentful/mappers/mapLandingInfoBlock.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import type { LandingPageInfo } from 'contentful/types'
+import type { MarketingTemplateInfoBoxBlock } from 'features/marketing-layouts/types'
+import { MarketingTemplateBlocks } from 'features/marketing-layouts/types'
+
+export const mapLandingInfoBlock = (blockItem: LandingPageInfo): MarketingTemplateInfoBoxBlock => ({
```
[the remark this answers is not in the document]
Comment (piekczyk) [reply contentful/mappers/mapLandingInfoBlock.ts:6]: fixed

File: contentful/mappers/mapLandingPageBannerBlock.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import type { LandingPageBanner } from 'contentful/types'
+import type { MarketingTemplateBannerBlock } from 'features/marketing-layouts/types'
+import { MarketingTemplateBlocks } from 'features/marketing-layouts/types'
+
+export const mapLandingPageBannerBlock = (
```
[the remark this answers is not in the document]
Comment (piekczyk) [reply contentful/mappers/mapLandingPageBannerBlock.ts:6]: fixed

File: contentful/mappers/mapLandingPageBenefitBlock.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import type { LandingPageBenefit } from 'contentful/types'
+import type { MarketingTemplateBenefitBoxBlock } from 'features/marketing-layouts/types'
+import { MarketingTemplateBlocks } from 'features/marketing-layouts/types'
+
+export const mapLandingPageBenefitBlock = (
```
[the remark this answers is not in the document]
Comment (piekczyk) [reply contentful/mappers/mapLandingPageBenefitBlock.ts:6]: fixed

File: contentful/mappers/mapLandingPageProductBlock.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
+import type { LandingPageProduct } from 'contentful/types'
+import type { MarketingTemplateProductBoxBlock } from 'features/marketing-layouts/types'
+import { MarketingTemplateBlocks } from 'features/marketing-layouts/types'
+
+export const mapLandingPageProductBlock = (
```
[the remark this answers is not in the document]
Comment (piekczyk) [reply contentful/mappers/mapLandingPageProductBlock.ts:6]: fixed

File: contentful/queries/getEntryCollection.ts [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+import { splitArrayToSameSizeChunks } from 'helpers/splitArrayToSameSizeChunks'
+
+const entryQuery = async (collectionIds: string[]) => {
+  const entry = await fetchGraphQL<EntryRawResponse>(`
+    query {
+      entryCollection(
+        where: {
+          sys: { id_in: ${JSON.stringify(collectionIds)} }
+        }
+        limit: 50
```
[the remark this answers is not in the document]
Comment (piekczyk) [reply contentful/queries/getEntryCollection.ts]: fixed

**Review by piotrkonowrocki — DISMISSED** [review]

**Review by piotrkonowrocki — DISMISSED** [review]

**Review by piotrkonowrocki — APPROVED** [review]

**Pull request merged**